src/renderers/renderer.util.tsx

```diff
@@ -121,16 +122,34 @@ export const registerStartupRenderer = (tester: RankedTester, renderer: any) =>
   return renderer;
 };
 
+const isRequired = (schema: JsonSchema, schemaPath: string): boolean => {
+    const pathSegments = schemaPath.split('/');
+    const lastSegment = pathSegments[pathSegments.length - 1];
+    const nextHigherSchemaSegments = pathSegments.slice(0, pathSegments.length - 2);
+    const nextHigherSchemaPath = nextHigherSchemaSegments.join('/');
+    const nextHigherSchema = resolveSchema(schema, nextHigherSchemaPath);
+
+    return nextHigherSchema !== undefined
+        && nextHigherSchema.required !== undefined
+        && nextHigherSchema.required.indexOf(lastSegment) !== -1;
+};
+
+export const setLabelField = (label: string, required: boolean): string => {
```

Comment:
what does this method set? Doesn't it only return something?

Reply:
Yes, at this point I am not sure where to define label setting. Or is it ok to use a method that returns the edited version of label in the control? Maybe it has to be set in the control's constructor. Do you have any advice?

Reply:
I would simply rename the method, to describe what it does. Something like `computeLabel`.